marketplaceItems/tipJar/entityResources/entityClientScripts/tipJar_client.js

```diff
@@ -0,0 +1,401 @@
+    function getSecondBalance() {
+        request(options, getSecondBalanceCallBack);
+    }
+
+
+    // Callback that is run after we request the first balance
+    var TIME_TILL_SECOND_BALANCE_CHECK_TIMEOUT_MS = 2500;
+    var MAXIMUM_FIRST_BALANCE_RETRIES = 5;
+    var FIRST_BALANCE_CHECK_INTERVAL = 250;
+    var firstBalanceRetryCount = 0;
+    var receivedFirstBalance = false;
+    var firstBalance = null;
+    function getFirstBalanceCallBack(error, result) {
+        // If we have already received the first balance or we have reached past the retry count then
+        // return from this callback.  
+        if (receivedFirstBalance || firstBalanceRetryCount >= MAXIMUM_FIRST_BALANCE_RETRIES) {
+            return;
+        }
+
+        // There was an error of some kind so retry. 
+        if (error || result.status !== "success"){
+            firstBalanceRetryCount++;
+            Script.setTimeout(function(){
+                request(options, getFirstBalanceCallBack);
+            }, FIRST_BALANCE_CHECK_INTERVAL);
+            return;
+        }
+
+        // We got the balance so call the commerce api and start calling for the second balance 
+        firstBalance = result.data.balance;
+        firstBalanceRetryCount = 0;
+        receivedFirstBalance = true;
+
+        Script.setTimeout(function(){
+            getSecondBalance();
+        }, TIME_TILL_SECOND_BALANCE_CHECK_TIMEOUT_MS);
+    }
+
+
+    // Called after we got the first balance to see if the tip went through
+    var MAXIMUM_SECOND_BALANCE_RETRIES = 7;
+    var SECOND_BALANCE_CHECK_INTERVAL = 1000;
+    var secondBalanceRetryCount = 0;
+    var secondBalance = null;
+    function getSecondBalanceCallBack(error, result) {
+        // The payment went through or we have hit the maximum amount of tries so return and reset balance related data
+        if (didThePaymentGoThrough() || secondBalanceRetryCount >= MAXIMUM_SECOND_BALANCE_RETRIES) {
+            resetBalanceChecks();
+            return;
+        }
+
+        if (error || result.status !== "success"){
+            secondBalanceRetryCount++;
+            Script.setTimeout(function(){
+                request(options, getSecondBalanceCallBack);
+            }, SECOND_BALANCE_CHECK_INTERVAL);
+            return;
+        }
+
+        secondBalance = result.data.balance;
+        if (didThePaymentGoThrough()) {
+            // The balance has gone through correctly.  Reset ur balance data and start feedback animation.
+            resetBalanceChecks();
+            startFeedback();
+            return;
+        }
+
+        // Second balance isn't the right amount, trying again
+        secondBalanceRetryCount++;
+        Script.setTimeout(function(){
+            request(options, getSecondBalanceCallBack);
+        }, SECOND_BALANCE_CHECK_INTERVAL);
+    }
+
+
+    // Check to see if the payment went through by examining the two balances
+    function didThePaymentGoThrough(){
```

Comment:
is there a reason this whole function can't be simplified to:
`return secondBalance === targetBalance;` ?